opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -184,6 +185,42 @@ func DiffSlice(leftSlice, rightSlice []string) []string {
 	return diff
 }
 
+// Count the number of pods running and ready and not terminating for a given nodePool
+func CountRunningPodsForNodePool(ctx context.Context, k8sClient client.Client, cr *opsterv1.OpenSearchCluster, nodePool *opsterv1.NodePool) (int, error) {
```

Comment:
do you think that func should work for all NodePools ? for example i dont know if it needs for clients nodePool

Reply:
In theory the logic could be more lenient for nodepools that are neither cluster_manager nor data. But I think it's better to prioritize cluster stability.